src/Jarvis.Addin.StackExchange/Common/QuestionDescriptionFactory.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Linq;
+using System.Text;
+using Jarvis.Core;
+
+namespace Jarvis.Addin.StackExchange.Common
+{
+    public interface IQuestionDescriptionFactory
+    {
+        string Create(Question question);
+    }
+
+    public class QuestionDescriptionFactory : IQuestionDescriptionFactory
+    {
+        private const string AnonymousUser = "anonymous";
+
+        public string Create(Question question)
+        {
+            Ensure.NotNull(question, nameof(question));
+
+            var description = new StringBuilder();
+
+            description.Append($"Asked {GetUserFriendlyDate(question.CreationDate)}");
+            if (string.Equals(question.Owner.DisplayName, AnonymousUser, StringComparison.CurrentCultureIgnoreCase))
+            {
+                description.Append($"by {question.Owner.DisplayName}");
+            }
+            description.Append(". ");
+
+            description.Append($"{question.AnswerCount} answer");
+            if (question.AnswerCount != 1)
+            {
+                description.Append("s");
+            }
+            if (question.AnswerCount != 0)
+            {
+                description.Append(question.IsAnswered ? ", including one " : ", none ");
+                description.Append("marked as correct");
+            }
+            description.Append(". ");
+
+            description.Append($"{question.Tags.Count} tag");
+            if (question.Tags.Count != 1)
+            {
+                description.Append("s");
+            }
+            if (question.Tags.Any())
+            {
+                description.Append(", including ");
+
+                var firstTag = question.Tags.FirstOrDefault();
+                var secondTag = question.Tags.Skip(1).FirstOrDefault();
+                var thridTag = question.Tags.Skip(2).FirstOrDefault();
+
+                description.Append($"'{firstTag}'");
+                if (!string.IsNullOrEmpty(secondTag))
+                {
+                    description.Append($", '{secondTag}'");
+                }
+                if (!string.IsNullOrEmpty(thridTag))
+                {
+                    description.Append($" and '{thridTag}'");
+                }
+                description.Append(".");
+            }
+
+            return description.ToString();
+        }
+
+        internal static string GetUserFriendlyDate(DateTime date)
```

Comment:
Isn't there some library for it?

Reply:
@reinismu I don't think we need to take on a dependency for this.